### ISSUE TITLE
Patch Fails Unless Manually Include MonoMod.Common

### ISSUE THREAD

**Issue by adstep** [pardeike/Harmony#409]
**Describe the bug**
The published NuGet package silently fails to apply a method patch. Built the project locally. With debug symbols, I was able to see that there was an error occurring where the library was unable to load ```MonoMod.Common```. Upon manually including that in my project, the patch was able to be applied - for both the locally built + NuGet package versions of the library.

Exception:
HarmonyLib.HarmonyException: 'Patching exception in method virtual System.Reflection.MethodBase System.Diagnostics.StackFrame::GetMethod()'

InnerException:
FileNotFoundException: Could not load file or assembly 'MonoMod.Common, Version=21.1.12.5, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

StackTrace:
   at HarmonyLib.HarmonySharedState.WithState[T](Func`1 action) in C:\dev\github\Harmony\Harmony\Internal\HarmonySharedState.cs:line 66
   at HarmonyLib.HarmonySharedState.GetPatchInfo(MethodBase method) in C:\dev\github\Harmony\Harmony\Internal\HarmonySharedState.cs:line 106
   at HarmonyLib.PatchClassProcessor.ProcessPatchJob(Job job) in C:\dev\github\Harmony\Harmony\Public\PatchClassProcessor.cs:line 180

**To Reproduce**
1. Clone [my repo](https://github.com/adstep/harmony-patch-failure)

Patch can be found in ```StackFramePatcher.cs``` and the patched method can be found in ```Greeter.cs```.

2. To see the silent failure, run the console application ```HarmonyPatchFailure.Demo```
3. To see my workaround, run the console application ```HarmonyPatchFailure.DemoFixed```

**Expected behavior**
Library should patch method without having to include MonoMod.Common directly.

**Runtime environment (please complete the following information):**
 - OS: Windows 10 64-bit
 - .NET 5
 - Harmony version >= 2.0.4
- Name of game or host application - N/A

**Comment by pardeike** [comment]
Did you execute the merge step by creating a release? Just compiling won’t do that which results in you having to include the dependency. Please check the project file to see that a merge step is part of the build.

**Comment by adstep** [comment]
@pardeike I modified the build to use ILRepack on the Debug build and continued to see the same error.

```
<When Condition="!$(IsCoreOrStandard)">
```

Is there some way I can turn on some debug logs w/ the NuGet version to confirm I'm seeing the same problem?

**Comment by pardeike** [comment]
The NuGet release does not support .NET 5 yet. So I suggest that after you build your Debug build you use a decompiler to see if the “missing” classes from MonoMod.Common are actually in the dll or not. 

Also: it was a while ago I updated the dependency version on MonoMod.Common. You can check (project has a comment to the releases page) if there is a newer version of MononMod.Common and use it.

**Comment by adstep** [comment]
I verified that MonoMod.Common isn't in the dll for .Net 5, which is a result of the condition on ILRepack in the csproj (Harmony.csproj:50).

```
<When Condition="!$(IsCoreOrStandard)">
```

I tried removing that condition but came across a few more errors building the project relating to ```ILRepack.Lib.MSBuild.targets```

![image](https://user-images.githubusercontent.com/7879048/123325777-4d0d7f80-d4ed-11eb-8f44-3d0c966cbeed.png)

Seems related to [this open issue](https://github.com/gluck/il-repack/issues/285) over on il-repack.

Not sure what the project supports, but it seems like this would impact .Net Core 3.0/3.1. I've verified the same problem occurs in my example for .Net Core 3.1, which makes sense given the condition. Manually adding the latest version of MonoMod.Common resolves my issue, again, however there might be other problems lurking since it looks like there are a few things normally packed that would be missing.

![image](https://user-images.githubusercontent.com/7879048/123327988-e0e04b00-d4ef-11eb-8246-11c71e95194b.png)

Thanks for your assistance in tracking this down.
